pandas/core/indexes/datetimes.py

```diff
@@ -186,7 +186,10 @@ class DatetimeIndex(DatelikeOps, TimelikeOps, DatetimeIndexOpsMixin,
     copy  : bool
         Make a copy of input ndarray
     freq : string or pandas offset object, optional
-        One of pandas date offset strings or corresponding objects
+        One of pandas date offset strings or corresponding objects. The string
+        'infer' can be passed in order to allow users to set the frequency of
```

Comment:
I think this would be a little more direct if the "to allow users" is removed, so it just reads as "...in order to set the frequency...".